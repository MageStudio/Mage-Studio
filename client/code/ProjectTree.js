import React from 'react';
import { Col, Tree, Icon } from 'antd';

const { TreeNode, DirectoryTree } = Tree;

import '../app/sidebars/sidebar.scss';

const ProjectTree = ({ scripts, onScriptSelect = f => f, config }) => {

    const { list = [] } = scripts;
    const {Â project } = config;

    return (
        <Col
            span={4}
            className="sidebar code-sidebar">
            <p className="title">
                <Icon className="icon" type="project" />
                <span>Project</span>
            </p>
            <div className='content'>
                <DirectoryTree
                    defaultExpandAll
                    onSelect={onScriptSelect}>
                    <TreeNode
                        className={"project-file"}
                        icon={<Icon
                            type="book"
                            className={"code-icon"}/>}
                        title={project}>
                        { list.map(s => (
                            <TreeNode
                                className={"project-file"}
                                icon={<Icon
                                    className="code-icon"
                                    type="file-text" />}
                                title={s.name}
                                key={s.name}
                                isLeaf />
                        ))}
                    </TreeNode>
                </DirectoryTree>
            </div>
        </Col>
    );
};

export default ProjectTree;
