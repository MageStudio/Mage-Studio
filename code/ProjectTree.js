import React from 'react';
import { Col, Tree, Icon } from 'antd';

const { TreeNode, DirectoryTree } = Tree;

import '../app/sidebars/sidebar.scss';

const ProjectTree = ({ scripts, onScriptSelect = f => f, config }) => {

    const { list = [] } = scripts;
    const {Â project } = config;

    return (
        <Col
            span={4}
            className="sidebar">
            <p className="title">
                <Icon className="icon" type="project" />
                <span>Project</span>
            </p>
            <div className='content'>
                <DirectoryTree
                    defaultExpandAll
                    onSelect={onScriptSelect}>
                    <TreeNode
                        icon={<Icon type="book" />}
                        title={project}>
                        { list.map((s, i) => (
                            <TreeNode
                                icon={<Icon type="file-text" />}
                                title={s.name}
                                key={s.name}
                                isLeaf />
                        ))}
                    </TreeNode>
                </DirectoryTree>
            </div>
        </Col>
    );
};

export default ProjectTree;
