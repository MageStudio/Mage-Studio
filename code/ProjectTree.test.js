import React from 'react';
import { shallow } from 'enzyme';

import ProjectTree from './ProjectTree';

const defaultProps = {
    scripts: {
        list: []
    },
    config: {
        project: 'test'
    }
};

describe('ProjectTree', () => {

    it('should render fine', () => {
        const component = shallow(<ProjectTree { ...defaultProps } />);
        expect(component).toMatchSnapshot();
    });
});
