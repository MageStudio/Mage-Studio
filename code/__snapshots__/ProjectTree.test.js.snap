// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectTree should render fine 1`] = `
<Col
  className="sidebar code-sidebar"
  span={4}
>
  <p
    className="title"
  >
    <Icon
      className="icon"
      type="project"
    />
    <span>
      Project
    </span>
  </p>
  <div
    className="content"
  >
    <DirectoryTree
      defaultExpandAll={true}
      expandAction="click"
      onSelect={[Function]}
      showIcon={true}
    >
      <TreeNode
        className="project-file"
        icon={
          <Icon
            className="code-icon"
            type="book"
          />
        }
        title={
          <ContextMenu>
            <span>
              test
            </span>
          </ContextMenu>
        }
      />
    </DirectoryTree>
  </div>
</Col>
`;
